src/components/pages/FavoriteFoods.tsx

```diff
@@ -1,79 +1,106 @@
+import { DragDropContext, Draggable, Droppable } from '@hello-pangea/dnd'
 import Box from '@mui/material/Box'
+import Button from '@mui/material/Button'
 import Container from '@mui/material/Container'
-import Stack from '@mui/material/Stack'
 import Typography from '@mui/material/Typography'
-import { useContext } from 'react'
-import { DragDropContext, Draggable } from 'react-beautiful-dnd'
-import { Link } from 'react-router-dom'
+import { useContext, useState } from 'react'
+import { Link, useParams } from 'react-router-dom'
+import { FavoriteFood } from '../../client/flavorite'
 import useFavorites from '../../hooks/useFavorites'
 import useUpdateFavorites from '../../hooks/useUpdateFavorites'
-import { StrictModeDroppable } from '../helpers/strictModeDroppable'
 import AddFavorite from '../partials/AddFavorite'
 import Spinner from '../partials/Spinner'
 import { UserContext } from '../partials/UserContext'
 
 export default function FavoriteFoods() {
   const { currentUser } = useContext(UserContext)
-  const username = currentUser!.username
+  const { username } = useParams()
+  const profileUsername = username!
   const {
     favorites,
     loading: loadingFavorites,
     error: errorFavorites,
-  } = useFavorites({ username: username })
+  } = useFavorites({ username: profileUsername })
   const { error: errorUpdateFavorites, mutate: updateFavorites } = useUpdateFavorites()
+  const [favsList, setFavsList] = useState<FavoriteFood[]>(favorites)
 
-  const favoritesList = favorites.map(({ id, name }, idx) => {
-    return (
-      <Draggable key={id} draggableId={`${id}`} index={idx}>
-        {(provided) => (
-          <Stack
-            spacing={2}
-            ref={provided.innerRef}
-            {...provided.draggableProps}
-            {...provided.dragHandleProps}
-          >
-            <Typography>
-              {id}.<Link to={`/${username}/favorites/${name}`}>{name}</Link>
-            </Typography>
-          </Stack>
-        )}
-      </Draggable>
-    )
-  })
+  const handleSelection = () => {
+    // TODO styling view review button based on mouse hover or click
+    console.log('review pop up')
+  }
 
-  const handleUpdateFavorites = (result: any) => {
-    const items = Array.from(favorites)
+  const handleUpdateFavorites = async (result: any) => {
+    const items = Array.from(favsList)
     const [reorderedItem] = items.splice(result.source.index, 1)
     items.splice(result.destination.index, 0, reorderedItem)
     items.forEach((item, idx) => {
       item.id = idx + 1
     })
```

Comment:
Array.from(favsList) is soft copy. means that it would change the original favsList data when we do 
`items.forEach((item, idx) => {
      item.id = idx + 1
    })`
as part of functional programming, we should not touch the original data otherwise it would have unexpected impact or bug. You can replace the data instead of touching the original object like this
`items = items.map((item, idx) => ({ ...item, id: idx+1 }))`

Reply:
*Also made this change in next PR unitTestFive*

I just changed the copy to ```const items = [...favsList]``` to clone the array, does this work? I didn't know if the splice methods were affecting the original array as well, so looked for a way to clone instead of softcopy. lemme know if this doens't work/isn't preferred

Reply:
Thanks. I made the change in unitTestFive as well, but I just saw you already approved that PR.
here's the code and confirmed that this way favsList is maintained as original.

```    let items = Array.from(favsList)
    const [reorderedItem] = items.splice(result.source.index, 1)
    items.splice(result.destination!.index, 0, reorderedItem)
    items = items.map((item, idx) => ({ ...item, id: idx + 1 }))```